lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb

```diff
@@ -137,35 +137,55 @@ def embedded_content_settings
         #         the pod targets.
         #
         def settings_to_import_pod_targets
-          if target.requires_frameworks?
-            build_pod_targets = pod_targets.select(&:should_build?)
-            framework_header_search_paths = build_pod_targets.map do |target|
-              "#{target.build_product_path}/Headers"
+          header_search_paths = []
+          silenced_header_search_paths = []
+          silenced_frameworks = []
+          pod_targets.each do |target|
+            target_header_search_paths = if target.should_build? && target.requires_frameworks?
+                                           ["#{target.build_product_path}/Headers"]
+                                         else
+                                           target.sandbox.public_headers.search_paths(target.platform)
+                                         end
+            header_search_paths += target_header_search_paths
+            if target.inhibit_warnings?
+              silenced_header_search_paths += target_header_search_paths
+              if target.requires_frameworks? && target.should_build?
+                silenced_frameworks.push(target.configuration_build_dir)
+              end
+            end
+          end
+
+          header_search_paths.uniq!
+          silenced_header_search_paths.uniq!
+
+          build_settings = {
+            # TODO: remove quote imports in CocoaPods 2.0
+            # Make headers discoverable with any syntax (quotes, brackets,
+            # @import, etc.)
+            'HEADER_SEARCH_PATHS' => XCConfigHelper.quote(header_search_paths),
```

Comment:
this looks like we'll be adding header search paths even for those things we're adding manually in `OTHER_CFLAGS` ?

Reply:
Yes, to support all import syntaxes. `HEADER_SEARCH_PATHS` makes headers discoverable by `#import "..."`, `#import <...>` and `@import ...`, whereas `-isystem` only works with the `#import <...>` syntax.

So with this patch the `-isystem` and `-iframework` flags we're adding to `HEADER_SEARCH_PATHS ` are not strictly needed for the code to compile, they are just added to silence warnings.